core/src/test/java/org/apache/brooklyn/core/entity/EntityConfigTest.java

```diff
@@ -244,59 +252,152 @@ public void testGetConfigMapWithSubValueAsStringNotCoerced() throws Exception {
     // of the previous "test.confMapThing.obj".
     //
     // Presumably an earlier call to task.get() timed out, causing it to cancel the task?
+    // Alex: yes, a task.cancel is performed for maps in
+    // AbstractEntity$BasicConfigurationSupport(AbstractConfigurationSupportInternal).getNonBlockingResolvingStructuredKey(ConfigKey<T>)    
+ 
+    //
     // I (Aled) question whether we want to support passing a task (rather than a 
     // DeferredSupplier or TaskFactory, for example). Our EntitySpec.configure is overloaded
     // to take a Task, but that feels wrong!?
-    @Test(groups="Broken")
-    public void testGetTaskNonBlocking() throws Exception {
-        final CountDownLatch latch = new CountDownLatch(1);
-        Task<String> task = Tasks.<String>builder().body(
+    //
+    // If starting clean I (Alex) would agree, we should use TaskFactory. However the
```

Comment:
Interesting idea that it is a single (shared) task.  I feel like each caller gets its own instance/subscription hence `TaskFactory`.  But for another day... (and in any case I'd love us to move to a JS promise/then model)